exonum-java-binding/qa-service/src/main/java/com/exonum/binding/qaservice/QaServiceImpl.java

```diff
@@ -130,6 +138,21 @@ public void createPublicApiHandlers(Node node, Router router) {
     controller.mountApi(router);
   }
 
+  @Override
+  public void beforeTransaction(BlockchainData blockchainData) {
+    // fixme! ECR-3849
+    HashCode counterId = Hashing.sha256()
+        .hashString(BEFORE_TXS_COUNTER_NAME, UTF_8);
+    incrementCounter(counterId, blockchainData);
+  }
+
+  @Override
+  public void afterTransactions(BlockchainData blockchainData) {
+    HashCode counterId = Hashing.sha256()
```

Comment:
extract method `private static HashCode getCounterId(String)`?

Reply:
I hope to get rid of it soonish, as ProofMaps now support String keys: ECR-3849